pkg/policy/repository.go

```diff
@@ -369,3 +370,33 @@ func (p *Repository) NumRules() int {
 func (p *Repository) GetRevision() uint64 {
 	return p.revision
 }
+
+// ConvertToK8sServiceToToCIDR traverses all egress rules and matches them against
+// provided serviceInfo. If a matching egress rule is found it is populated
+// with ToCIDR entries based on endpoint object.
+func (p *Repository) ConvertToK8sServiceToToCIDR(serviceInfo types.K8sServiceNamespace, endpoint types.K8sServiceEndpoint) error {
```

Comment:
I would suggest to move the mutex lock here. I know you need to delete and then add the toCIDR while holding the lock but I would suggest to do that in a new function called "Replace", or something like that, so you can lock/unlock the mutex inside the repository.go file